docs/walkthrough.md

```diff
@@ -46,91 +46,78 @@ forc parse-bytecode out/debug/dbg_example.bin
+          1   4      NOOP                                  47 00 00 00
+          2   8      InvalidOpcode                         00 00 00 00   data section offset lo (0)
+          3   12     InvalidOpcode                         00 00 00 44   data section offset hi (68)
+          4   16     LW { ra: 63, rb: 12, imm: 1 }         5d fc c0 01
+          5   20     ADD { ra: 63, rb: 63, rc: 12 }        10 ff f3 00
+          6   24     MOVE { ra: 18, rb: 1 }                1a 48 10 00
+          7   28     MOVE { ra: 17, rb: 0 }                1a 44 00 00
+          8   32     LW { ra: 16, rb: 63, imm: 0 }         5d 43 f0 00
+          9   36     LT { ra: 16, rb: 17, rc: 16 }         16 41 14 00
+         10   40     JNZI { ra: 16, imm: 13 }              73 40 00 0d   conditionally jump to byte 52
+         11   44     LOG { ra: 18, rb: 0, rc: 0, rd: 0 }   33 48 00 00
+         12   48     RET { ra: 0 }                         24 00 00 00
+         13   52     ADD { ra: 17, rb: 17, rc: 1 }         10 45 10 40
+         14   56     MUL { ra: 18, rb: 18, rc: 17 }        1b 49 24 40
+         15   60     JI { imm: 8 }                         90 00 00 08   jump to byte 32
+         16   64     NOOP                                  47 00 00 00
+         17   68     InvalidOpcode                         00 00 00 00
+         18   72     InvalidOpcode                         00 00 00 05
 ```
 
-We can recognize the `while` loop by the conditional jumps `JNZI(..)`. The condition just before the first jump can be identified by `LT(..)` instruction (for `<`). Some notable instructions that are generated only once in our code include `MUL(..)` for multiplication and `LOG(.., 0, 0, 0)` from the `log` function.
+We can recognize the `while` loop by the conditional jumps `JNZI`. The condition just before the first jump can be identified by `LT` instruction (for `<`). Some notable instructions that are generated only once in our code include `MUL` for multiplication and `LOG {.., 0, 0, 0}` from the `log` function.
 
 ## Setting up the debugging
 
-We can start up the debug infrastructure. On a new terminal session run `fuel-core --db-type in-memory`; we need to have that running because it actually executes the program. Now we can fire up the debugger itself: `fuel-debugger`. Now
+We can start up the debug infrastructure. On a new terminal session run `fuel-core --db-type in-memory --debug`; we need to have that running because it actually executes the program. Now we can fire up the debugger itself: `fuel-debugger`. Now
 if everything is set up correctly, you shoould see the debugger prompt (`>> `). You can use `help` command to list available commands.
 
 Now we would like to inspect the program while it's running. To do this, we first need to send the script to the executor, i.e. `fuel-core`. To do so, we need a *transaction specification*, `tx.json`. It looks something like this:
 
 ```json
```

Comment:
Is this file generated somehow, or are users meant to create it by hand?

Reply:
The JSON file is created by hand. Maybe I should write the instructions to do that somewhere, but currently the process is quite cumbersome. I'm hoping to build a tool soon.